### ISSUE TITLE
Need help: Recommended usage of the framework to implement Scrollable ListView

### ISSUE THREAD

**Issue by tim37021** [pureqml/qmlcore#163]
I am really newbie to this framework. A lot thanks for bringing QML to the Web, thought I need to add more components for my regular use(it's very easy and straightforward). anyway, our designers like it!!

I've got couple of questions here! I am implementing scrollable listview. I am not using listview here because it hides everything out of contrainer...

```qml
           CssMixin {
               rules: ".no-scrollbar {.....}";
           }

            ScrollView {
                id: bpm_view;
                anchors.fill: parent;

                ClassMixin {
                    classList: ["no-scrollbar"];
                }

                Repeater {
                    model: 20;
                    delegate: Rectangle { 
                        y: model.index*bpm_view.height;
                        width: bpm_view.width;
                        height: bpm_view.height;
                        **property int bpm: 5*model.index+60;**
                        Text {
                            anchors.verticalCenter: parent.verticalCenter;
                            anchors.horizontalCenter: parent.horizontalCenter;
                            text: parent.bpm;
                            font.family: "微軟正黑體";
                            color: "white";

                        }
                    }
                    
                }
            }

```

Question: 
1. ClassMixin is a hack for inserting additional classes into the classList of div. Is there any more elegant way for doing this?
2. CssMixin seems to add style element to the document.  I've read the core framework code. I don't get how it prevents re-adding same style if reusing the XXX component multiple time.

XXX.qml: 
```qml
CssMixin {
rules: ".cls {.....}";
}
```

app.qml:
```qml
XXX {
}

XXX {
}
```
3. Could I use template literal?
Text {
  text: `${object.name} is good`;
}

4. bold line has bug if changed to...
model.index*5+60;

5. For the ugly scrollable listview implementation I list above. It's there any better choice to implement this widget? I also want to know which item is selected..

**Comment by whoozle** [comment]
First of all, it's preferable to use ListView, instead of Repeater, this will give you selected index, proper focus, keyboard and mouse navigation. 

There's also nativeScrolling property defined in BaseView.qml. If you turn it on, you will get native scrollbars and smooth mobile scrolling. 

Unfortunately we don't detect type of scrollings automatically at the moment, because all people need different behaviour here. 

Regarding styling,
Instead of using css declaration, we're using slightly different technique to register classes and styles. 

Each Item provides a way of register styles, see documentation in Item.qml 

```
	///@private
	function registerStyle(style, tag) {
		var rules = '*** your css code here ***'
		style.addRule(tag, rules)
	}

	/// returns tag for corresponding element
	function getTag() { return 'div' } //you can override html tag here

	/// returns tag for corresponding element
	function getClass() { return '' } //you can override class used

```

Hope this helps! Have fun :)

**Comment by tim37021** [comment]
@whoozle 
Hey I've implement Connection component because it's so importance for writing a elegant QML code

The following is my implement. If it is okay would you accept a pull request?
```qml
Object {
    property EventEmitter target;
    onCompleted: {
        
        if(this.target != null) {
            for(let [ename, callbacks] of Object.entries(this._eventHandlers)) {
                for(let callback of callbacks)
                    this.target.on(ename, callback);
            }
            for(let [ename, property_storage] of Object.entries(this.__properties)) {
                for(let callback of property_storage.onChanged)
                    this.target.onChanged(ename, callback);
            }
            
        }else {
            log('Warning! no target')
        }
    }
}
```

**Comment by tim37021** [comment]
I wonder how to add multiple classes to an element
function getClass() { return 'MYCLASS1 MYCLASS2' }

**Comment by whoozle** [comment]
Could you please create another issue for class? There's no simple solution to this, probably we could allow array of classes to be returned there. 

Currently we're not supportnig multiple classes, and normally your css code can clash with our css code. Can you just pass your styles to registerStyle and/or CssMixin. 


Regarding connections, you can perfectly connect to remote objects in our dialect, e.g.

```
mybutton.onClicked: { log("hello"); }
```

Connection is only useful when you switching the targets, and it's not really trivial to implement, but since there's multiple request for it, I'll probably do it soon.

**Comment by whoozle** [comment]
Implemented Connection in e5bd88fc134ddcad27177fa19a3353d9d36e1208

**Comment by tim37021** [comment]
> Implemented Connection in [e5bd88f](https://github.com/pureqml/qmlcore/commit/e5bd88fc134ddcad27177fa19a3353d9d36e1208)

Thanks!
